pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -329,6 +329,24 @@ type ClusterCephxConfig struct {
 	// Daemon configures CephX key settings for local Ceph daemons managed by Rook and part of the
 	// Ceph cluster. Daemon CephX keys can be rotated without affecting client connections.
 	Daemon CephxConfig `json:"daemon,omitempty"`
+
+	// CSI configures CephX key rotation settings for the Ceph-CSI daemons in the current Kubernetes cluster.
+	// CSI key rotation can affect existing PV connections, so take care when exercising this option.
+	CSI CephXConfigWithPriorCount `json:"csi,omitempty"`
+}
+
+type CephXConfigWithPriorCount struct {
+	CephxConfig `json:",inline"` // inline core CephxConfig
+
+	// PriorKeyCount available for components that use overlapping key rotation or
+	// for non-local Ceph daemon eg. CSI, RBD/CephFS mirror.
+	// This tells Rook how many prior keys to keep active.
+	// Generally, this would be set to 1 to allow for a migration period for applications.
+	// If desired, set this to 0 to delete prior keys after migration
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=10
+	KeepPriorKeyCount uint32 `json:"KeepPriorKeyCount,omitempty"`
```

Comment:
Nit: json vals should begin with lower case

```suggestion
	KeepPriorKeyCount uint32 `json:"keepPriorKeyCount,omitempty"`
```